tests/utilities/test_attributes.py

```diff
@@ -0,0 +1,90 @@
+from random import random
+
+import pytest
+from phoenix.utilities.attributes import unflatten
+
+
+@pytest.mark.parametrize(
+    "key_value_pairs,desired",
+    [
+        ((), {}),
+        ((("1", 0),), {"1": 0}),
+        ((("1.2", 0),), {"1": {"2": 0}}),
+        ((("1.0.2", 0),), {"1": [{"2": 0}]}),
+        ((("1.0.2.3", 0),), {"1": [{"2": {"3": 0}}]}),
+        ((("1.0.2.0.3", 0),), {"1": [{"2": [{"3": 0}]}]}),
+        ((("1.0.2.0.3.4", 0),), {"1": [{"2": [{"3": {"4": 0}}]}]}),
```

Comment:
Is there some ambiguity here? How do I know that 4 is a key and not an index?

Reply:
Yea, it is ambiguous. So in our methodology, 4 isn't considered an index here because we don't flatten lists unless each element is a dict, i.e. there are more keys to follow.